testing/spectest/shared/common/forkchoice/builder.go

```diff
@@ -119,9 +128,13 @@ func (bb *Builder) Check(t testing.TB, c *Check) {
 		return
 	}
 	ctx := context.TODO()
+	bb.service.ForkChoicer().Lock()
 	require.NoError(t, bb.service.UpdateAndSaveHeadWithBalances(ctx))
+	bb.service.ForkChoicer().Unlock()
```

Comment:
this can be deferred

Reply:
I don't think I can defer it because `bb.service.CurrentJustifiedCheckpt` requires a lock :(